crates/viewer/re_viewer_context/src/test_context.rs

```diff
@@ -67,52 +80,51 @@ impl TestContext {
     pub fn run(&self, mut func: impl FnMut(&ViewerContext<'_>, &mut egui::Ui)) {
         egui::__run_test_ctx(|ctx| {
             egui::CentralPanel::default().show(ctx, |ui| {
-                re_ui::apply_style_and_install_loaders(ui.ctx());
-                let blueprint_query = LatestAtQuery::latest(blueprint_timeline());
-
-                let component_ui_registry = ComponentUiRegistry::new(Box::new(
-                    |_ctx, _ui, _ui_layout, _query, _db, _entity_path, _row_id, _component| {},
-                ));
-
-                let store_context = StoreContext {
-                    app_id: "rerun_test".into(),
-                    blueprint: &self.blueprint_store,
-                    default_blueprint: None,
-                    recording: &self.recording_store,
-                    bundle: &Default::default(),
-                    caches: &Default::default(),
-                    hub: &Default::default(),
-                };
-
-                let rec_cfg = RecordingConfig::default();
-                rec_cfg.time_ctrl.write().set_timeline(self.active_timeline);
-
-                let egui_context = ui.ctx().clone();
-                let ctx = ViewerContext {
-                    app_options: &Default::default(),
-                    cache: &Default::default(),
-                    reflection: &Default::default(),
-                    component_ui_registry: &component_ui_registry,
-                    space_view_class_registry: &self.space_view_class_registry,
-                    store_context: &store_context,
-                    applicable_entities_per_visualizer: &Default::default(),
-                    indicated_entities_per_visualizer: &Default::default(),
-                    query_results: &Default::default(),
-                    rec_cfg: &rec_cfg,
-                    blueprint_cfg: &Default::default(),
-                    selection_state: &self.selection_state,
-                    blueprint_query: &blueprint_query,
-                    egui_ctx: &egui_context,
-                    render_ctx: None,
-                    command_sender: &self.command_sender,
-                    focused_item: &None,
-                };
-
-                func(&ctx, ui);
+                let egui_ctx = ui.ctx().clone();
+
+                self.run_simple(&egui_ctx, |ctx| {
+                    func(ctx, ui);
+                });
             });
         });
     }
 
+    pub fn run_simple(&self, egui_ctx: &egui::Context, func: impl FnOnce(&ViewerContext<'_>)) {
```

Comment:
This should have a docstring. I also don't understand the naming.

It's more like the previous `run` should be called `run_ui`, and this `run_test` or just `run`?

Reply:
Does this even need to be `pub`?

Reply:
I'm reshuffling/improving things here. I'm expecting that this will further evolve when we better integrate this with kittest.

Reply:
> Does this even need to be pub?

Yeah, since we use it in re_time_panel

Reply:
I think we have:
- `run` (does just the basic thing, takes a `FnOnce(&ViewerContex)`
-  `run_in_egui_central_panel (legacy thing that use `__run_test_ctx` and takes `FnMut(&ViewerContext, Ui)`—should probably go away in the future
- some form of a `run_with_kittest()` (for a future PR)

All of these should `handle_system_command()`, so we can get rid of `run_and_handle_system_commands()`.

I'm updating this PR according to this.